community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/commands/predicates/Predicate.scala

```diff
@@ -175,7 +175,7 @@ trait StringOperator {
     case (null, _) => None
     case (_, null) => None
     case (l: String, r: String) => Some(compare(l,r))
-    case (l, r) => throw new CypherTypeException(s"Expected two strings, but got $l and $r")
+    case (_, _) => None
```

Comment:
Why not have the string-string case, and then a default handler that returns None?